libphobos/libdruntime/core/threadasm.S

```diff
@@ -28,6 +28,15 @@
 .section .note.GNU-stack,"",%progbits
 #endif
 
+/* Let preprocessor tell us if C symbols have a prefix: __USER_LAEBL_PREFIX__ */
+#ifdef __USER_LABEL_PREFIX__
+#define GLUE2(a, b) a ## b
+#define GLUE(a, b) GLUE2(a, b)
```

Comment:
It is a case of the non-intuitive (at least to me) cpp macro expansion rule where a layer of indirection is needed to expand macros used with `##`.
http://stackoverflow.com/questions/1489932/c-preprocessor-and-token-concatenation